tests/integration/minicluster_v2.go

```diff
@@ -0,0 +1,261 @@
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"sync"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.uber.org/zap"
+
+	grpcdatacoord "github.com/milvus-io/milvus/internal/distributed/datacoord"
+	grpcdatacoordclient "github.com/milvus-io/milvus/internal/distributed/datacoord/client"
+	grpcdatanode "github.com/milvus-io/milvus/internal/distributed/datanode"
+	grpcdatanodeclient "github.com/milvus-io/milvus/internal/distributed/datanode/client"
+	grpcindexnode "github.com/milvus-io/milvus/internal/distributed/indexnode"
+	grpcindexnodeclient "github.com/milvus-io/milvus/internal/distributed/indexnode/client"
+	grpcproxy "github.com/milvus-io/milvus/internal/distributed/proxy"
+	grpcproxyclient "github.com/milvus-io/milvus/internal/distributed/proxy/client"
+	grpcquerycoord "github.com/milvus-io/milvus/internal/distributed/querycoord"
+	grpcquerycoordclient "github.com/milvus-io/milvus/internal/distributed/querycoord/client"
+	grpcquerynode "github.com/milvus-io/milvus/internal/distributed/querynode"
+	grpcquerynodeclient "github.com/milvus-io/milvus/internal/distributed/querynode/client"
+	grpcrootcoord "github.com/milvus-io/milvus/internal/distributed/rootcoord"
+	grpcrootcoordclient "github.com/milvus-io/milvus/internal/distributed/rootcoord/client"
+	"github.com/milvus-io/milvus/internal/storage"
+	"github.com/milvus-io/milvus/internal/util/dependency"
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/milvus-io/milvus/pkg/util/etcd"
+	"github.com/milvus-io/milvus/pkg/util/paramtable"
+)
+
+type MiniClusterV2 struct {
+	ctx context.Context
+
+	mu sync.RWMutex
+
+	params        map[string]string
+	clusterConfig ClusterConfig
+
+	factory      dependency.Factory
+	ChunkManager storage.ChunkManager
+
+	EtcdCli *clientv3.Client
+
+	Proxy      *grpcproxy.Server
+	DataCoord  *grpcdatacoord.Server
+	RootCoord  *grpcrootcoord.Server
+	QueryCoord *grpcquerycoord.Server
+
+	DataCoordClient  *grpcdatacoordclient.Client
+	RootCoordClient  *grpcrootcoordclient.Client
+	QueryCoordClient *grpcquerycoordclient.Client
+
+	ProxyClient     *grpcproxyclient.Client
+	DataNodeClient  *grpcdatanodeclient.Client
+	QueryNodeClient *grpcquerynodeclient.Client
+	IndexNodeClient *grpcindexnodeclient.Client
+
+	DataNode  *grpcdatanode.Server
+	QueryNode *grpcquerynode.Server
+	IndexNode *grpcindexnode.Server
+
+	MetaWatcher MetaWatcher
+}
+
+type OptionV2 func(cluster *MiniClusterV2)
+
+func StartMiniClusterV2(ctx context.Context, opts ...OptionV2) (*MiniClusterV2, error) {
+	cluster := &MiniClusterV2{
```

Comment:
It seems can't execute tests in parallel with this MinClusterV2？

Reply:
I think it can? It use different etcd and storage rootpath

Reply:
I forget the port conflicts, add a logic to get available ports for each test. I think it can solve most parallel problems